### PR TITLE
[DIM] Throwing AmbiguousImplementationException on call of multiple candidate implementations.

### LINKED ISSUE #13926: [DIM] [CoreCLR Test] diamondshape not working
IL Example: 
https://github.com/dotnet/coreclr/blob/bbd79053fffa3bee7ce06df30115f077c077082a/tests/
src/Loader/classloader/DefaultInterfaceMethods/diamondshape/diamondshape.il

### PR DESCRIPTION
**Pull request by thaystg**
Adding AmbiguousImplementationException exception and throwing this when there are multiple candidate implementations.
Fixes #13926

### CODE REVIEW

File: mono/utils/mono-error.c [hunk 1]
```diff
@@ -437,6 +437,21 @@ mono_error_set_not_supported (MonoError *oerror, const char *msg_format, ...)
...
+ * mono_error_set_ambiguous_implementation:
+ *
+ * System.AmbiguousImplementationException
+ */
+void
+mono_error_set_ambiguous_implementation (MonoError *oerror, const char *msg_format, ...)
+{
+	va_list args;
+	va_start (args, msg_format);
+	mono_error_set_generic_errorv (oerror, "System.Runtime", "AmbiguousImplementationException", msg_format, args);
```
Comment (lambdageek) [inline mono/utils/mono-error.c]: I think you also need to update the [linker descriptor](https://github.com/mono/mono/blob/master/mcs/class/corlib/LinkerDescriptor/mscorlib.xml) to ensure that the `AmbiguousImplementationException` class is preserved.

Comment (filipnavara) [reply mono/utils/mono-error.c]: Pleas update the [netcore one](https://github.com/mono/mono/blob/master/mcs/class/System.Private.CoreLib/LinkerDescriptor/System.Private.CoreLib.xml) too. Thanks!

Comment (thaystg) [reply mono/utils/mono-error.c]: Done! :)

**Comment by marek-safar** [comment]
This also needs to be backported to 2019-04

**Comment by thaystg** [comment]
@monojenkins commit csproj

**Comment by thaystg** [comment]
@monojenkins commit apidiff

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Pull request merged**